addons/mail/static/src/core/common/thread_model.js

```diff
@@ -312,6 +312,8 @@ export class Thread extends Record {
     });
     /** @type {"not_fetched"|"pending"|"fetched"} */
     fetchMembersState = "not_fetched";
+    /** @type {integer|null} */
+    highlightMessageId = null;
```

Comment:
This refers to a message, would be best as a relational field `Record.one("Message")`.
This makes the code easier to understand, as it's easier to deal with record themselves than properties on record. This doesn't seem much between `message` and `messageId`, but for thread it's the difference between `thread` and `threadId, threadModel`... Which is clear that you prefer a single naturally cohesive object (with briefer name) than 2 heavily coupled but non-cohesive props that are less readable to read
(1 word variable vs 2 words variable in camelCase becomes quickly unreadable)

Reply:
I would still need to keep the `highlightMessageId` in the `formRenderer` and pass it to chatter. I don't think it's a good idea to create the message record in `formRenderer` because the certainty of the thread being available in the setup is not there.
Because of that, I will need to add a `useState`, but I am not sure whether the router will have the `highlight_message_id` as it is not going to be saved in its state.